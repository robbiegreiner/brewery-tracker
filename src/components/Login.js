import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import firebase, { auth, provider } from '../firebase.js';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      user: null
    };
  }

  login() {
    auth.signInWithPopup(provider)
      .then((result) => {
        const user = result.user;
        this.setState({
          user
        });
      });
  }

  render() {
    return (
      <div className='login'>
        <h1>Login Here</h1>
        <input type='text' name='username'></input>
        <input type='text' name='password'></input>
        <button onClick={this.login.bind(this)}>Log In</button>
      </div>
    );

  }
}

export default Login;
