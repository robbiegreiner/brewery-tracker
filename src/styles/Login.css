.login {
  margin-top: 120px; }

.login .error-message {
  color: #FF644E; }

.login {
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #27394A;
  height: 400px;
  width: 400px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 10px #bdbdbd; }

.login h2 {
  text-align: center;
  color: white; }

.login input {
  margin-top: 30px; }

.password-input,
.email-input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 20px 0;
  color: white; }

.password-input:focus,
.email-input:focus {
  outline: none;
  border-bottom: 2px solid #D23321; }

.login-btn {
  background-color: #FF644E;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 8px 20px;
  font-size: 16px;
  margin: 20px 0;
  width: 100%;
  cursor: pointer; }

.form-button:hover {
  background-color: #aa2f1e; }

.form-button:disabled {
  background-color: #bdbdbd; }

.error {
  margin-top: 10px;
  color: darkgray; }

.login a {
  color: white; }

@media screen and (max-width: 550px) {
  .login {
    width: 300px; } }
