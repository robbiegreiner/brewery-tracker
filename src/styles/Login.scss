.login {
  margin-top: 120px;
}
