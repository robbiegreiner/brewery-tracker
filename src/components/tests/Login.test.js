import React from "react";
import Login from '../Login';
import { shallow } from "enzyme";

describe('Login component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<Login
    user={{id:'vafahiuweh'}}
    getBrewery={mockFn}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
