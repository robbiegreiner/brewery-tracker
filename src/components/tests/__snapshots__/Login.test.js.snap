// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    errorMessage=""
    login={[Function]}
    user={
        Object {
            "id": "vafahiuweh",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Redirect
          push={false}
          to="/"
/>,
        <h1>
          Login
</h1>,
        "",
        <input
          className="email-input"
          onChange={[Function]}
          placeholder="email"
/>,
        <input
          className="password-input"
          onChange={[Function]}
          placeholder="password"
          type="password"
/>,
        <button
          className="login-btn"
          onClick={[Function]}
>
          submit
</button>,
        <Link
          replace={false}
          to="/createaccount"
>
          <h4>
                    Not a member? Create an account!
          </h4>
</Link>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h1",
      },
      "",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "email-input",
          "onChange": [Function],
          "placeholder": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "password-input",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "submit",
          "className": "login-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h4>
            Not a member? Create an account!
</h4>,
          "replace": false,
          "to": "/createaccount",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Not a member? Create an account!",
          },
          "ref": null,
          "rendered": "Not a member? Create an account!",
          "type": "h4",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Redirect
            push={false}
            to="/"
/>,
          <h1>
            Login
</h1>,
          "",
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="email"
/>,
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="password"
            type="password"
/>,
          <button
            className="login-btn"
            onClick={[Function]}
>
            submit
</button>,
          <Link
            replace={false}
            to="/createaccount"
>
            <h4>
                        Not a member? Create an account!
            </h4>
</Link>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h1",
        },
        "",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "onChange": [Function],
            "placeholder": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "submit",
            "className": "login-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h4>
              Not a member? Create an account!
</h4>,
            "replace": false,
            "to": "/createaccount",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Not a member? Create an account!",
            },
            "ref": null,
            "rendered": "Not a member? Create an account!",
            "type": "h4",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
